app/controllers/comments_controller.rb

```diff
@@ -1,32 +1,31 @@
 class CommentsController < ApplicationController
-  before_action :find_article, only: %i[create]
   before_action :find_comment, only: %i[destroy]
-
   http_basic_authenticate_with name: 'user', password: 'secret', only: :destroy
 
   def create
-    @comment = @article.comments.create(comments_params)
-    redirect_to article_path(@article)
+    @comment = Comment.new(comments_params)
+    @comment.article_id = params[:article_id]
```

Comment:
You can build the params hash, including the `article_id`, in the comments_params method. That way, you don't need to assign the `article_id` to  `@comment`.

Reply:
I made this operation and returning a hash, is correct doing this ?
```ruby
def comments_params
    comment = params.require(:comment).permit(:commenter, :body, :status).to_h
    comment['article_id'] = params['article_id']
    comment
end
```

Reply:
Update from `comment_params`
```ruby
def comment_params
    comment = params.require(:comment).permit(:commenter, :body, :status)
    comment.merge(article_id: params[:article_id])
end
```